qp_klp/klp.py

```diff
@@ -158,3 +161,138 @@ def sequence_processing_pipeline(qclient, job_id, parameters, out_dir):
     paths = [(f'{final_results_path}/', 'directory')]
     return (True, [ArtifactInfo('output', 'job-output-folder', paths)],
             status_line.msg)
+
+
+def prep_NuQCJob(qclient, job_id, parameters, out_dir):
+    """Sequence Processing Pipeline command
+
+    Parameters
+    ----------
+    qclient : tgp.qiita_client.QiitaClient
+        The Qiita server client
+    job_id : str
+        The job id
+    parameters : dict
+        The parameter values for this job
+    out_dir : str
+        The path to the job's output directory
+
+    Returns
+    -------
+    bool, list, str
+        The results of the job
+    """
+    status_line = StatusUpdate(qclient, job_id)
+
+    status_line.update_job_status("Setting up pipeline")
+
+    pid = parameters.pop('prep_id')
+
+    prep_info = qclient.get(f'/qiita_db/prep_template/{pid}/')
+    dt = prep_info['data_type']
+    sid = prep_info['study']
+    if dt not in {'Metagenomic', 'Metatranscriptomic'}:
+        return (False, [], f'Prep {pid} has a not valid data type: {dt}')
+    aid = prep_info['artifact']
+    if not str(aid).isnumeric():
+        return (False, [], f'Prep {pid} has a not valid artifact: {aid}')
+
+    files, pt = qclient.artifact_and_preparation_files(aid)
+    pt.set_index('sample_name', inplace=True)
+
+    out_path = partial(join, out_dir)
+    final_results_path = out_path('final_results')
+    makedirs(final_results_path, exist_ok=True)
+
+    project_name = f'qiita-{pid}-{aid}_{sid}'
+
+    sheet = MetagenomicSampleSheetv90()
+    sheet.Header['IEMFileVersion'] = '4'
+    sheet.Header['Date'] = datetime.today().strftime('%m/%d/%y')
+    sheet.Header['Workflow'] = 'GenerateFASTQ'
+    sheet.Header['Application'] = 'FASTQ Only'
+    sheet.Header['Assay'] = prep_info['data_type']
+    sheet.Header['Description'] = f'prep_NuQCJob - {pid}'
+    sheet.Header['Chemistry'] = 'Default'
+    sheet.Header['SheetType'] = 'standard_metag'
+    sheet.Header['SheetVersion'] = '90'
+    sheet.Header['Investigator Name'] = 'Qiita'
+    sheet.Header['Experiment Name'] = project_name
+
+    sheet.Bioinformatics = pd.DataFrame(
+        columns=['Sample_Project', 'ForwardAdapter', 'ReverseAdapter',
+                 'library_construction_protocol',
+                 'experiment_design_description',
+                 'PolyGTrimming', 'HumanFiltering', 'QiitaID'],
+        data=[[project_name, 'NA', 'NA', 'NA', 'NA', 'FALSE', 'TRUE', sid]])
```

Comment:
Is there a reason this stuff isn't done by the `MetagenomicSampleSheetv90` constructor?